src/server/game/Entities/Player/Player.cpp

```diff
@@ -16440,12 +16441,18 @@ bool Player::HasQuestForItem(uint32 itemid) const
 
         QuestStatusData const& q_status = qs_itr->second;
 
-        if (q_status.Status == QUEST_STATUS_INCOMPLETE)
+        if (q_status.Status == QUEST_STATUS_INCOMPLETE || q_status.Status == QUEST_STATUS_COMPLETE)
```

Comment:
Including QUEST_STATUS_COMPLETE here allows quest items to drop even if player completes the objective, see the use of Player::HasQuestForItem in LootItem::AllowedForPlayer